pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -2414,7 +2423,7 @@ func parentRefString(ref k8s.ParentReference) string {
 		ptr.OrEmpty(ref.Namespace))
 }
 
-// buildHostnameMatch generates a VirtualService.spec.hosts section from a listener
+// buildHostnameMatch generates a Gateway.spec.servers.hosts section from a listener
```

Comment:
This comment appears to have been inaccurate - the VirtualService.spec.hosts section would be built from HTTPRoute.spec.hostnames - [Listener](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.Listener) configuration is used to build the Istio Gateway servers.